### DISCUSSION TITLE
Doubts regarding changing menus

### DISCUSSION THREAD

**Discussion by PedroQuesado** [axmolengine/axmol#2213] [Q&A] [answered]
i thinking creating a game and have this flow menu of start ->character selection->not have character created can create a 
character ->character selection -> world
Doubts its i create a scene for menu of start, character selection,character creation and world or create a new class of this and extends from the main class?
i think the way the optimize the code and not creating alot of garbage in the memory
what its the best way

**Comment by aismann** [comment]
> what its the best way

Have  a look on `cpp-tests `examples

**Reply by PedroQuesado** [reply]
i saw a quick look from the cpp-tests
i want your opinion about a best away to creating the menu of start, character selection,character creation and world without creating too much garbage in memory
to be honest i java developer not a c++

**Reply by rh101** [reply]
> i want your opinion about a best away to creating the menu of start, character selection,character creation and world without creating too much garbage in memory
> to be honest i java developer not a c++

That's a general game structure/design issue, one that you should be able to find a multitude of resources for on the internet.  It's not something specific to Axmol or any other game engine.  

Also, not quite sure what you mean by "too much garbage in memory".  

If you haven't already started working on the project, then do so, and as a personal opinion, address issues as you come across them, and don't try to prematurely optimize things.  As you have stated, you are coming from a Java development background, so best to just dive in and become more familiar with C++.

**Comment by PedroQuesado** [comment]
thanks for your opinion
about "too much garbage in memory" its for example variable its not be used

**Reply by rh101** [reply]
> about "too much garbage in memory" its for example variable its not be used

I still don't quite understand, since things like variables are completely under your control, so I'm not entirely sure how anyone can help with how you decide to write your code.

**Comment by PedroQuesado** [comment]
ok i gonna study the game structure/design  
i java /spring boot developer
i know java to create micro services
but thanks for your help

**Comment by aismann** [comment]
@PedroQuesado 
You mean the `Garbage Collection` in Java?
C++ has no `Garbage Collection` native.
You have the full control,

**Reply by PedroQuesado** [reply]
yes

**Reply by aismann** [reply, marked as answer]
Axmol cant give you full answers to C++ questions. Thats not the focus here.
Plz ask on a C++ forum.
Also you can have a look on keywords like` Scope/Lifetime of Variables`

@PedroQuesado 
Your title gives a wrong look on this entry.
Plz change it or close it what Im prefer.
